libcxx/include/__atomic/atomic_sync.h

```diff
@@ -102,21 +135,19 @@ _LIBCPP_HIDE_FROM_ABI bool __cxx_nonatomic_compare_equal(_Tp const& __lhs, _Tp c
   return std::memcmp(std::addressof(__lhs), std::addressof(__rhs), sizeof(_Tp)) == 0;
 }
 
-template <class _Atp, class _Tp>
-struct __cxx_atomic_wait_test_fn_impl {
-  _Atp* __a;
+template <class _Tp>
+struct __cxx_atomic_wait_poll_fn_impl {
```

Comment:
I would suggest renaming this to `__atomic_compare_equal_to` or something like that -- that's still not a great name but it is a bit more descriptive, and in the future we could probably write this as a lambda (which is what would really make this code easier to read).

Reply:
Yes, a lambda would definitely help. I renamed this struct to `__atomic_compare_unequal_to`. So the logic can be read as "wait unless/until the atomic variable compares unequal to val" (similar to the wording in the spec: <https://eel.is/c++draft/atomics#ref.ops-23.2>).